code/game/machinery/telecomms/telecomunications.dm

```diff
@@ -548,8 +547,8 @@ var/global/list/obj/machinery/telecomms/telecomms_list = list()
 /obj/machinery/telecomms/server/proc/admin_log(var/mob/mob)
 	var/msg="[key_name(mob)] has compiled a script to server [src]:"
 	diary << msg
-	diary << rawcode
-	src.investigate_log("[msg]<br>[rawcode]", "ntsl")
+	diary << rawcode.Join("")
```

Comment:
You mentioned the admin_log is quite costly, so only do the ```.Join()``` once, store it in a local variable to this function, and use it for both the ```<<``` and the ```inv_log()```.